### PR TITLE
common: misc fixes detected by crypto shutdown assert

### PR DESCRIPTION
**Pull request by liewegas**
These are fixes uncovered by #12624, without the assert on shutdown that
helped find them (since not all leaks have been found yet).

### CODE REVIEW

**Comment by yuriw** [comment]
test this please

**Comment by yuriw** [comment]
test this please

File: src/global/signal_handler.cc [hunk 1]
```diff
@@ -111,17 +111,19 @@ static void handle_fatal_signal(int signum)
...
   pidfile_remove();
 
+  // TODO: don't use an ostringstream here. It could call malloc(), which we
+  // don't want inside a signal handler.
+  // Also fix the backtrace code not to allocate memory.
+  BackTrace bt(0);
+  ostringstream oss;
+  bt.print(oss);
+  dout_emergency(oss.str());
+
```
Comment (badone) [inline src/global/signal_handler.cc]: Just a thought here. Would this code benefit from Kefu's PR 9028 ?

**Comment by liewegas** [comment]
Not in this case, I think, because dout_emergency takes a string?

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-01-13_21:05:46-rados-wip-yuri-testing2_2017_1_13-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-01-14_16:29:03-rados-wip-yuri-testing2_2017_1_13---basic-smithi/

(see http://tracker.ceph.com/issues/18478)

centos run
http://pulpito.front.sepia.ceph.com/yuriw-2017-01-14_16:34:35-rados-wip-yuri-testing2_2017_1_13-distro-basic-vps/

**Pull request merged**